### PR TITLE
Only one instance for each (host+port, client, user)

### PR DESCRIPTION
**Pull request by johan-bjareholt**
Have not tested on windows

Also updated test_client.py for the new aw-core versions

Doesn't fix https://github.com/ActivityWatch/aw-qt/issues/19, but at least makes it less severe

### CODE REVIEW

**Comment by ErikBjare** [comment]
What happens if the program is killed by SIGTERM and the __del__ clause is never run? 

Doesn't look like we inform the user where the lock file is in case it prevents the program from starting.

**Comment by johan-bjareholt** [comment]
On linux it still works, the lock is released by the kernel.

I have not tested on windows, but should work since it tries to remove the lock file and the lock on the file should be released by the kernel when it is exiting.

Also just relized that you currently cannot run both testing and non-testing at the same time, will fix that.

**Comment by ErikBjare** [comment]
I think the lock file should probably be in the cache dir instead of the data dir, or what do you think?

I think the testing vs non-testing should really depend on the `module_id + host + port` the client tries to connect to, not `module_id + testing`.

**Comment by johan-bjareholt** [comment]
Regarding if it should be in the cache dir instead of data dir I agree, will fix.

Regarding testing vs non-testing i currently disagree. Since host+port is fetched from the config file anyway there is little point in doing it like that. If for example you would run the clients as another user the lock files would be in another directory also so that doesn't become an issue. If you would want to report to two servers it would be better to simply utilize aw-sync for that once that's working.

**Comment by ErikBjare** [comment]
Strong disagree on removing the ability to send data to multiple hosts from a client.

**Just to clarify:** Testing will behave the same (pick a special host and port in the config) unless specified. The lockfile name should include `host+port` instead of a `testing` tag. Otherwise we would limit a watcher to two running instances in non-testing mode. Why important to avoid this limitation is explaned below.

**About sync:** The server instances listening to different hosts and ports might not want to sync. **Example:** one watcher sends to some data collection service (automatic sharing/status/analysis services, studies, etc.) and the other to a local server as the users own copy.

 `aw-sync`'s scope is *personal, private, sync*. Bucket-specific time-interval sync to third party sources is not within the scope of `aw-sync` (yet?). Would add quite a bit of complexity and require exposing the server, need authentication, etc.

The user might want to "go live" with their data using some service, adding a server sync would increase latency significantly making the reporting happen after 2 syncs. I don't think we should limit `aw-sync` to be the only syncing solution possible, connecting to multiple servers directly from the client is an excellent alternative for many uses. A sync tool gets too complicated for some uses when third parties are involved.

We should offer host and port keyword parameters on the ActivityWatchClient constructor. This is needed if we want to be able to specify host and port for a single run using cli arguments. So applications managing the watcher can set some settings temporarily without having to deal with the config file (and overriding settings set there).

**Comment by ErikBjare** [comment]
But we can merge this for now if you want. Are you done?

File: aw_client/__init__.py [hunk 1]
```diff
@@ -1,4 +1,2 @@
-from .config import load_config
-
-from .client import ActivityWatchClient
+from .client import ActivityWatchClient # noqa
```
Comment (ErikBjare) [inline aw_client/__init__.py]: What is `#noqa`?

Comment (johan-bjareholt) [inline aw_client/__init__.py]: Removed the "Import unused" from landscape.io

**Comment by johan-bjareholt** [comment]
> erb long wall of text

Fair enough. Your example was rather exotic but valid and the host+port solution is more flexible so let's do it that way.

The only issue now would be if someone tried to connect to "localhost:5600" and another client tries at "127.0.0.1:5600" the lock wouldn't work.

**Comment by ErikBjare** [comment]
I'll admit it might be a bit exotic. But imo it's such a large gain for such a small effort that it cannot be ignored.

File: aw_client/client.py [hunk 2]
```diff
@@ -36,8 +36,10 @@ def __init__(self, client_name: str, testing=False) -> None:
         client_config = load_config()
         configsection = "server" if not testing else "server-testing"
 
-        self.server_hostname = client_config[configsection]["hostname"]
-        self.server_port = client_config[configsection]["port"]
+        self.server_host = server_host if server_host != None else client_config[configsection]["hostname"]
+        self.server_port = server_port if server_port != None else client_config[configsection]["port"]
+
+        self.instance = SingleInstance("{}@{}:{}".format(self.client_name, self.server_host, self.server_port))
```
Comment (ErikBjare) [inline aw_client/client.py]: I don't think that will be a valid in a Windows file. Just use dashes in place of `@` and `:`.

Comment (ErikBjare) [reply aw_client/client.py]: This needs to be fixed.

**Pull request merged**